spec/services/idv/duplicate_ssn_finder_spec.rb

```diff
@@ -144,4 +182,76 @@
       end
     end
   end
+
+  describe '#associated_facial_match_profiles_with_ssn' do
+    let(:ssn) { '123-45-6789' }
+    let(:user) { create(:user) }
+    let(:user2) { create(:user) }
+    let(:sp) { 'urn:gov:gsa:openidconnect:inactive:sp:test' }
+    let(:identity) do
+      build(
+        :service_provider_identity,
+        service_provider: sp,
+        session_uuid: SecureRandom.uuid,
+        ial: 2,
+      )
+    end
+    let(:identity2) do
+      build(
+        :service_provider_identity,
+        service_provider: sp,
+        session_uuid: SecureRandom.uuid,
+        ial: 2,
+      )
+    end
+    subject { described_class.new(ssn: ssn, user: user, issuer: sp) }
+
+    before do
+      allow(IdentityConfig.store).to receive(:eligible_one_account_providers)
+        .and_return([
+                      'urn:gov:gsa:openidconnect:inactive:sp:test',
+                      'urn:gov:gsa:openidconnect:inactive:sp:test2',
+                    ])
+    end
+    context 'when ssn belongs to another profile with the same sp' do
+      it 'returns matching profile id' do
+        create(:profile, :facial_match_proof, id: 1, pii: { ssn: ssn }, user: user, active: true)
+        user.identities << identity
```

Comment:
Why have the identities associated only in this context and not in the sibling contexts? In fact, having these be associated in the next context should demonstrate that the code is wrong because the test fails.